### PR TITLE
Improve task name handling

### PR DESCRIPTION
**Pull request by aphedges**
Building off the work in #44, I've made some improvements to the handling of task names while hopefully maintaining backwards compatibility.

The new file `tasks.json` was generated using the files in [json_creation.zip](https://github.com/allenai/ScienceWorld/files/10348397/json_creation.zip):

- `task_names.json`: A file created for internal use from data in the paper and repository
- `convert.py`: A script to combine our JSON file and your `dict`s into a new file for inclusion in the library

I didn't actually test my changes by running ScienceWorld, but the tests make me pretty confident it would work.

### CODE REVIEW

File: scienceworld/scienceworld.py [hunk 1]
```diff
@@ -139,11 +134,11 @@ def tasks(self):
     @property
     def task_names(self):
         """ Get the name for the supported tasks in ScienceWorld. """
-        return list(ID2TASK.values())
+        return sorted(ID2TASK.values())
```
Comment (MarcCote) [inline scienceworld/scienceworld.py:137]: Why the `sorted` here?

File: scienceworld/utils.py [hunk 2]
```diff
@@ -10,7 +10,10 @@ def infer_task(name_or_id):
     if name_or_id in ID2TASK:
         name_or_id = ID2TASK[name_or_id]
 
-    # Remove prefix "task-##-" from task name.
-    name_or_id = re.sub(r"task-(\d|a|b)+-", "", name_or_id)
+    # Correct typo fixed in b807f742050ba5d9e0c5483624c39834368cd34f
+    name_or_id = name_or_id.replace("mendellian", "mendelian")
```
Comment (MarcCote) [inline scienceworld/utils.py:14]: Good thinking.

File: scienceworld/scienceworld.py (hunk 1 above)
Comment (MarcCote) [reply scienceworld/scienceworld.py:137]: Ok, I see later with the unit test, it is to make sure Python and Scala (i.e., getTaskNames) returns the same thing. I think I'd prefer the tasks to be listed as they appear in the paper. That's why I was using an OrderedDict for ID2Task.

**Review by MarcCote — CHANGES_REQUESTED** [review]
Thanks for the clean up. I like the external json file to store the available tasks. @PeterAJansen any thoughts in this PR?

**Comment by PeterAJansen** [comment]
In retrospect, it seems like referencing the tasks by either the ID in the paper (e.g. "3-1") or their detailed name (e.g. "create-a-circuit") is the way to go, since it makes sure there's no ambiguity.  The 0-29 numbering that's auto-sorted seemed great in development, but seems less great in retrospect. :)

File: scienceworld/scienceworld.py (hunk 1 above)
Comment (aphedges) [reply scienceworld/scienceworld.py]: @MarcCote, that makes sense! I've moved the sorting to the test. I briefly tried to have the Scala code output the same order, but it seemed like more effort than it was worth.

**Review by MarcCote — APPROVED** [review]

**Pull request merged**